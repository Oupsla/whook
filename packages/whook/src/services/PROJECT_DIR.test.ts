import YError from 'yerror';

describe('initProjectDir', () => {
  const pkgDir = jest.fn();
  const log = jest.fn();

  beforeEach(() => {
    jest.resetModules();
    pkgDir.mockReset();
    log.mockReset();
  });

  it('should use the env port first', async () => {
    jest.doMock('pkg-dir', () => pkgDir);
    pkgDir.mockResolvedValueOnce('/home/whoiam/projects/my-whook/project');

    const initProjectDir = require('./PROJECT_DIR').default;
    const PROJECT_DIR = await initProjectDir({
      PWD: '/home/whoiam/projects/my-whook/project/src/lol',
      log,
    });

    expect({
      PROJECT_DIR,
      pkgDirCalls: pkgDir.mock.calls,
      logCalls: log.mock.calls.filter(args => 'stack' !== args[0]),
    }).toMatchSnapshot();
  });

  it('should find a port by itself if no env port', async () => {
    jest.doMock('pkg-dir', () => pkgDir);
    pkgDir.mockResolvedValueOnce('');

    const initProjectDir = require('./PROJECT_DIR').default;
    try {
      await initProjectDir({
        PWD: '/home/whoiam/documents',
        log,
      });
      throw new YError('E_UNEXPECTED_SUCCESS');
    } catch (err) {
      expect({
        errorCode: err.code,
        errorParams: err.params,
        pkgDirCalls: pkgDir.mock.calls,
        logCalls: log.mock.calls
          .filter(args => 'stack' !== args[0])
          .map(([arg1, arg2, ...args]) => {
            return [arg1, arg2.replace(/port (\d+)/, 'port ${PORT}'), ...args];
          }),
      }).toMatchSnapshot();
    }
  });
});
