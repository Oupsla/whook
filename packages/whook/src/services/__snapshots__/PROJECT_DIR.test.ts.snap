// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initProjectDir should find a port by itself if no env port 1`] = `
Object {
  "errorCode": "E_NO_PROJECT_DIR",
  "errorParams": Array [
    "/home/whoiam/documents",
  ],
  "logCalls": Array [
    Array [
      "error",
      "ðŸš« - Could not detect the whook project dir.",
    ],
  ],
  "pkgDirCalls": Array [
    Array [
      "/home/whoiam/documents",
    ],
  ],
}
`;

exports[`initProjectDir should use the env port first 1`] = `
Object {
  "PROJECT_DIR": "/home/whoiam/projects/my-whook/project",
  "logCalls": Array [
    Array [
      "warning",
      "âœ” - Found the project dir:",
      "/home/whoiam/projects/my-whook/project",
    ],
  ],
  "pkgDirCalls": Array [
    Array [
      "/home/whoiam/projects/my-whook/project/src/lol",
    ],
  ],
}
`;
